backend/resources/center/main.js

```diff
@@ -0,0 +1,88 @@
+//@ts-check
+
+(function () {
+    const vscode = acquireVsCodeApi();
+
+    const oldState = vscode.getState() || { guides: [] };
+
+    /** @type {Array<{ extensionId: string, id: string, title: string, description: string, disabled?: boolean }>} */
+    let guides = oldState.guides;
+    updateGuideList(guides);
+
+    // Handle messages sent from the extension to the webview
+    window.addEventListener('message', event => {
+        const message = event.data;
+        switch (message.type) {
+            case 'refreshData':
+                {
+                    guides = message.data.items;
+                    updateGuideList(guides);
+                    break;
+                }
+        }
+    });
+
+    /**
+     * @param {Array<{ extensionId: string, id: string, title: string, description: string, disabled?: boolean }>} guides
+     */
+    function updateGuideList(guides) {
+        const ul = document.querySelector('.guide-list');
```

Comment:
It seems the `ul` will be referenced in 62nd, 70th, 75th lines and etc. Do we want to make some existence checking on this var? E.g.:
- if it is a nullish value, the iteration logic below would be affected.
- if it exists, some `?.` operators in lines below could be omitted?

Reply:
As above.